dulwich/client.py

```diff
@@ -997,13 +997,22 @@ def default_urllib2_opener(config):
 
 class HttpGitClient(GitClient):
 
-    def __init__(self, base_url, dumb=None, opener=None, config=None, **kwargs):
+    def __init__(self, base_url, dumb=None, opener=None, config=None,
+                 username=None, passwd=None, **kwargs):
         self._base_url = base_url.rstrip("/") + "/"
         self.dumb = dumb
         if opener is None:
             self.opener = default_urllib2_opener(config)
         else:
             self.opener = opener
+
+        if passwd is None:
+            passwd = ""
```

Comment:
No password specified is not the same as no password; you probably to check for 'username is not None and password is not None' on line 1012.